x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/ldap/PoolingSessionFactorySettings.java

```diff
@@ -35,6 +36,26 @@
                     key -> secureString(key, null)
             );
 
+    public static final Function<String, Setting.AffixSetting<String>> BIND_MODE = RealmSettings.affixSetting("bind.mode",
+            key -> new Setting<>(key, "simple", Function.identity(), v -> {
+                switch (v) {
+                case "simple":
+                case "sasl_gssapi":
+                    break;
+                default:
+                    throw new IllegalArgumentException("only [simple] and [sasl_gssapi] bind mode are allowed, [" + v + "] is invalid");
+                }
+            }, Setting.Property.NodeScope));
+
+    public static final Function<String, Setting.AffixSetting<String>> SASL_GSSAPI_PRINCIPAL = RealmSettings
+            .affixSetting("sasl_gssapi.bind.principal", key -> Setting.simpleString(key, Setting.Property.NodeScope, Property.Filtered));
```

Comment:
why `Property.Filtered` the principal?

Reply:
Thanks for considering it. But, and I might be wrong here, aren't hostnames a common principal name in kerberos service configuration? In this case it does not make sense to hide it. I have no experience with Kerberos deployments, if you say it's filtered then it's filtered.